lemming.go

```diff
@@ -55,36 +54,35 @@ func New(lis net.Listener, opts ...grpc.ServerOption) (*Device, error) {
 		p4rtServer:  fp4rt.New(s),
 	}
 	reflection.Register(s)
-	if err := d.startServer(); err != nil {
-		return nil, fmt.Errorf("failed to start device: %v", err)
-	}
+	d.startServer()
 	return d, nil
 }
 
 // Addr returns the currently configured ip:port for the listening services.
 func (d *Device) Addr() string {
-	return d.addr
+	return d.lis.Addr().String()
 }
 
 // Stop stops the listening services.
-func (d *Device) Stop() {
-	if d.stop == nil {
-		return
-	}
+func (d *Device) Stop() error {
 	d.stop()
+	return d.err
 }
 
 // GNMI returns the gnmi server implementation.
 func (d *Device) GNMI() *fgnmi.Server {
 	return d.gnmiServer
 }
 
-func (d *Device) startServer() error {
-	go d.s.Serve(d.lis)
+func (d *Device) startServer() {
+	go func() {
+		d.err = d.s.Serve(d.lis)
+	}()
 
 	d.stop = func() {
 		d.s.Stop()
-		d.lis.Close()
+		if err := d.lis.Close(); err != nil {
+			klog.Info(err)
+		}
```

Comment:
curious why log the error instead of storing it?

Reply:
because we don't really care about why closing the listener failed vs. the server actually failing